scalyr_agent/metrics/functions.py

```diff
@@ -300,6 +307,25 @@ def calculate(
         # TODO: Use dataclass once we only support Python 3
         result = [(rate_metric_name, rate_value)]
 
+        # Periodically we run clean up routine to remove stale entries. We perform this on this
+        # function call which means it may occasionally add some overhead to this method call.
+        # If we did it in a background thread or similar this would complicate thread safety
+        # aspects and mean we would need to obtain a lock on every emit_value() call which may be
+        # even more expensive.
+        # TODO (longer term): To simplify and speed up things we should perhaps have an instance of
+        # RateMetricFunction per monitor. This would also help us avoid any thread safety issues
+        # (monitor already runs in a separate dedicated thread).
+        metric_functions_cleanup_interval = (
+            monitor._global_config
+            and monitor._global_config.metric_functions_cleanup_interval
+            or 0
+        )
+
+        if metric_functions_cleanup_interval > 0 and (
+            cls.LAST_CLEANUP_RUNTIME_TS <= (now_s - metric_functions_cleanup_interval)
+        ):
+            cls._remove_old_entries(monitor=monitor)
+
```

Comment:
Do you think we should have option to disable that cleanup by making interval zero? Or maybe at least log a warning message on zero?

Reply:
Yeah, if the interval is set to ``0``, it will be disabled. It's mostly there for testing purposes. We don't really plan to expose this functionality to the end user (config option is mostly there for us for testing and similar).